ambry-api/src/main/java/com.github.ambry/account/Account.java

```diff
@@ -0,0 +1,312 @@
+   */
+  public Container getContainerByName(String containerName) {
+    return containerNameToContainerMap.get(containerName);
+  }
+
+  /**
+   * Gets all the containers of this account in a Collection.
+   * @return All the containers of this account.
+   */
+  public Collection<Container> getAllContainers() {
+    return Collections.unmodifiableCollection(containerIdToContainerMap.values());
+  }
+
+  /**
+   * Adds a {@link Container} to this account and updates internal maps accordingly.
+   * @param container The container to update this account.
+   */
+  private void updateContainerMap(Container container) {
+    containerIdToContainerMap.put(container.getId(), container);
+    containerNameToContainerMap.put(container.getName(), container);
+  }
+
+  /**
+   * Generates a String representation that uniquely identifies this account. The string is in the format of
+   * {@code Account[id]}.
+   * @return The String representation of this account.
+   */
+  @Override
+  public String toString() {
+    return "Account[" + getId() + "]";
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof Account)) {
+      return false;
+    }
+
+    Account account = (Account) o;
+
+    if (!id.equals(account.id)) {
+      return false;
+    }
+    if (!name.equals(account.name)) {
+      return false;
+    }
+    return status == account.status;
+  }
+
+  @Override
+  public int hashCode() {
+    return (int) id;
+  }
+
+  /**
+   * The status of the account. {@code ACTIVE} means this account is in operational state, and {@code INACTIVE} means
+   * the account has been deactivated.
+   */
+  public enum AccountStatus {
+    ACTIVE, INACTIVE
+  }
+
+  /**
+   * Checks if required fields are missing to build.
+   */
+  private void checkAccountIdInContainers() {
+    for (Container container : containerIdToContainerMap.values()) {
+      if (container.getParentAccountId() != id) {
+        throw new IllegalStateException(
+            "Container does not belong to this account because parentAccountId=" + container.getParentAccountId()
+                + " is not the same as accountId=" + id);
+      }
+    }
+  }
+
+  /**
+   * Checks if any required field is missing for a {@link Account}.
+   */
+  private void checkRequiredFieldsForBuild() {
+    if (id == null || name == null || status == null) {
+      throw new IllegalStateException(
+          "Either of required fields id=" + id + " or name=" + name + " or status=" + status + " is null");
+    }
+  }
+
+  /**
+   * Checks if there are containers that have different ids but the same names, or vise versa.
+   * @param containers A collection of containers to check duplicate name or id.
+   * @throws IllegalStateException if there are containers that have different ids but the same names, or vise versa.
+   */
+  private void checkDuplicateContainerNameOrId(Collection<Container> containers) {
```

Comment:
Could this method just take in the expectedNumberOfContainers? This means that we won't have to construct a new list when building an account from JSON metadata.